code/modules/bitrunning/server/obj_generation.dm

```diff
@@ -73,11 +73,21 @@
 	return wayout
 
 /// Scans over neo's contents for bitrunning tech disks. Loads the items or abilities onto the avatar.
-/obj/machinery/quantum_server/proc/stock_gear(mob/living/carbon/human/avatar, mob/living/carbon/human/neo)
+/obj/machinery/quantum_server/proc/stock_gear(mob/living/carbon/human/avatar, mob/living/carbon/human/neo, datum/lazy_template/virtual_domain/generated_domain)
+	var/domain_forbids_items = generated_domain.forbids_disk_items
+	var/domain_forbids_spells = generated_domain.forbids_disk_spells
 	var/failed = FALSE
 
+	if(domain_forbids_items)
+		to_chat(neo, span_warning("This domain forbids the use of smuggled digital equipment, your disk items will not be granted!"))
+	if(domain_forbids_spells)
+		to_chat(neo, span_warning("This domain forbids the use of imported abilities, your disk powers will not be granted!"))
```

Comment:
It would be nicer if only one message was sent if both are enabled.
Something like this could do but if you want to do it another way, that's cool too.
```
	var/import_ban = list()
	var/disk_ban = list()
	if(domain_forbids_items)
		import_ban += "smuggled digital equipment"
		disk_ban += "items"
	if(domain_forbid_spells)
		import_ban += "imported_abilities"
		disk_ban += "powers"
	if(import_ban )
		to_chat(neo, span_warning("This domain forbids the use of [english_list(import_ban], your disk [english_list(disk_ban)] will not be granted!"))
```